File: ce/snippets/csharp/CRMV8/outlook/cs/retrievedatafilters2.cs
// <snippetretrievedatafilters2>



 // Call InstantiateFiltersRequest
 Console.WriteLine("Retrieving user's ID and creating the template collection");
 WhoAmIRequest whoAmI = new WhoAmIRequest();
 Guid id = ((WhoAmIResponse)_serviceProxy.Execute(whoAmI)).UserId;
 EntityReferenceCollection templates = new EntityReferenceCollection();
 templates.Add(new EntityReference(
     SavedQuery.EntityLogicalName,
     _offlineTemplate));

 Console.WriteLine("Activating the selected offline templates for this user");
 InstantiateFiltersRequest request = new InstantiateFiltersRequest
 {
     UserId = id,
     TemplateCollection = templates                            
 };
 InstantiateFiltersResponse response =
     (InstantiateFiltersResponse)_serviceProxy.Execute(request);
 Console.WriteLine();

// </snippetretrievedatafilters2>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/outlookvb/vb/retrievedatafilters2.vb
' <snippetretrievedatafilters2>



 ' Call InstantiateFiltersRequest
 Console.WriteLine("Retrieving user's ID and creating the template collection")
 Dim whoAmI As New WhoAmIRequest()
 Dim id As Guid = (CType(_serviceProxy.Execute(whoAmI), WhoAmIResponse)).UserId
 Dim templates As New EntityReferenceCollection()
 templates.Add(New EntityReference(SavedQuery.EntityLogicalName, _offlineTemplate))

 Console.WriteLine("Activating the selected offline templates for this user")
 Dim request As InstantiateFiltersRequest = New InstantiateFiltersRequest With {.UserId = id, .TemplateCollection = templates}
 Dim response As InstantiateFiltersResponse = CType(_serviceProxy.Execute(request), InstantiateFiltersResponse)
 Console.WriteLine()

' </snippetretrievedatafilters2>